Assets/Shopify/UIToolkit/Shops/Generic/Classes/GenericSingleProductShop.cs

```diff
@@ -1,44 +1,194 @@
 ﻿namespace Shopify.UIToolkit.Shops {
     using Shopify.Unity;
     using Shopify.Unity.SDK;
+    using Shopify.UIToolkit;
     using UnityEngine;
+    using UnityEngine.UI;
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using System.Globalization;
 
     [RequireComponent(typeof(SingleProductShopController))]
     public class GenericSingleProductShop : MonoBehaviour, ISingleProductShop {
-        void IShop.OnError(ShopifyError error) {
-            throw new System.NotImplementedException();
+
+        [HeaderAttribute("Text Fields")]
+        public Text Title;
+        public Text Price;
+        public Text Description;
+
+        [HeaderAttribute("Variant Selection")]
+        public Dropdown VariantDropdownMenu;
+
+        [HeaderAttribute("Images")]
+        public GameObject ProductImageViewArea;
+        public ProductImageHolder ProductImageHolderTemplate;
+        public GameObject ProductImageContainer;
+        public RemoteImageLoader ActiveImage;
+
+        public ProductVariant CurrentSelectedVariant {
+            get {
+                if (_productVariants == null) {
+                    return null;
+                }
+
+                return _productVariants[VariantDropdownMenu.value];
+            }
+        }
+
+        private ProductVariant[] _productVariants;
+        private SingleProductShopController _controller;
+
+        private const string SingleVariantTitle = "Default Title";
+
+        #region Mono Behaviour
+        
+        void Awake() {
+            _controller = GetComponent<SingleProductShopController>();
+            _controller.Shop = this;
+            _controller.OnShow();
+        }
+
+        #endregion
+
+        #region IShop Interface
+
+        void IShop.OnLoadingStarted() {}
```

Comment:
Should these not just be left empty so the exceptions happen?

Reply:
I've put the exceptions back in